corehq/ex-submodules/casexml/apps/case/xml/parser.py

```diff
@@ -70,7 +70,7 @@ def get_known_properties(self):
                     if getattr(self, p) is not None)
 
     def __repr__(self):
-        return "CaseAction(block={})".format(self.raw_block)
+        return f"{type(self).__name__}(block={self.raw_block!r})"
```

Comment:
This could change the output here (the `!r` part), right. Probably nothing depends on that, but just wanted to point that out

Reply:
Technically it could, but it should not change for any practical values of `self.raw_block`. I believe `self.raw_block` is nearly always a `dict`, but if it were a string the old way would not show that while this new way will.

Reply:
Note however that `type(self).__name__` will almost certainly never be `CaseAction`. It will be the name of the a subclass of `CaseActionBase`. Something like `CaseUpdateAction`. This was actually the thing that prompted the change: I was in a debugger looking at a case actions, but I could not easily tell what type they were by looking at their representation.